pkg/cmd/roachtest/tests/large_schema_backup_restore.go

```diff
@@ -0,0 +1,122 @@
+		),
+		CompatibleClouds: registry.OnlyGCE,
+		Suites:           registry.Suites(registry.Weekly),
+		Timeout:          12 * time.Hour,
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			// Number of tables per-database from the TPCC template.
+			const numTablesPerDB = 100
+			numDbs := numTables/numTablesPerDB + 1
+			dbs := make([]string, numDbs)
+			tables := make([]string, numTables)
+
+			c.Start(ctx, t.L(), option.NewStartOpts(option.NoBackupSchedule), install.MakeClusterSettings(), c.CRDBNodes())
+			conn := c.Conn(ctx, t.L(), 1)
+			defer conn.Close()
+
+			// Create a user that can log in to the console
+			_, err := conn.ExecContext(ctx, "CREATE USER roachadmin password 'roacher'")
+			require.NoError(t, err)
+			_, err = conn.ExecContext(ctx, "GRANT ADMIN to roachadmin")
+			require.NoError(t, err)
+
+			t.L().Printf("Creating tables")
+			tx, err := conn.BeginTx(ctx, nil)
+			require.NoError(t, err)
+			for i := range numDbs {
+				t.L().Printf("Building database %d of %d", i+1, numDbs)
+				dbName := fmt.Sprintf("test_db_%d", i)
+				dbs[i] = dbName
+				_, err := tx.ExecContext(ctx, fmt.Sprintf("CREATE DATABASE %s", dbName))
+				require.NoError(t, err)
+
+				for j := range numTablesPerDB {
+					tableIndex := i*numTablesPerDB + j
+					if tableIndex >= numTables {
+						break
+					}
+					tableName := fmt.Sprintf("%s.kv_%d", dbs[i], j)
+					tables[tableIndex] = tableName
+					_, err := tx.ExecContext(ctx, fmt.Sprintf("CREATE TABLE %s (key STRING NOT NULL PRIMARY KEY, value STRING NOT NULL)", tableName))
+					require.NoError(t, err)
+				}
+			}
+			require.NoError(t, tx.Commit())
+
+			dest := destinationName(c)
+			uri := `gs://` + backupTestingBucket + `/` + dest + `?AUTH=implicit`
+			t.L().Printf("Backing up to %s\n", uri)
+			_, err = conn.ExecContext(ctx, "BACKUP INTO $1 WITH REVISION_HISTORY", uri)
+			require.NoError(t, err)
+
+			const numLayers = 24
+			for i := range numLayers {
+				t.L().Printf("Building layer %d of %d", i+1, numLayers)
+				tx, err = conn.BeginTx(ctx, nil)
+				require.NoError(t, err)
+				for _, tableName := range tables {
+					_, err := tx.ExecContext(ctx, fmt.Sprintf("DROP TABLE %s", tableName))
+					require.NoError(t, err)
+					_, err = tx.ExecContext(ctx, fmt.Sprintf("CREATE TABLE %s (key STRING NOT NULL PRIMARY KEY, value STRING NOT NULL)", tableName))
+					require.NoError(t, err)
+				}
+				require.NoError(t, tx.Commit())
+
+				_, err = conn.Exec("BACKUP INTO LATEST IN $1 WITH REVISION_HISTORY", uri)
+				require.NoError(t, err)
+
+			}
+
+			t.L().Printf("Dropping databases\n") // Needed to do the full cluster restore.
+			tx, err = conn.BeginTx(ctx, nil)
+			require.NoError(t, err)
+			for _, dbName := range dbs {
+				_, err = tx.Exec("DROP DATABASE " + dbName + " CASCADE")
+				require.NoError(t, err)
+			}
+			require.NoError(t, tx.Commit())
+
+			t.L().Printf("Restoring from %s\n", uri)
+			_, err = conn.Exec("RESTORE FROM LATEST IN $1", uri)
```

Comment:
nit: perhaps verify you did indeed restore 10k descriptors? something like `select count(*) from system.namespace` should be greater than 10k (given system tables)

Reply:
👍